controllers/restic.go

```diff
@@ -0,0 +1,122 @@
+package controllers
+
+import (
+	"fmt"
+
+	"github.com/go-logr/logr"
+	pvcv1alpha1 "github.com/konveyor/volume-snapshot-mover/api/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+// Restic secret data keys
+const (
+	// AWS vars
+	AWSAccessKey = "AWS_ACCESS_KEY_ID"
+	AWSSecretKey = "AWS_SECRET_ACCESS_KEY"
+
+	// TODO: GCP and Azure
+
+	// Restic repo vars
+	ResticPassword   = "RESTIC_PASSWORD"
+	ResticRepository = "RESTIC_REPOSITORY"
+)
+
+// Restic secret vars to create new secrets
+var (
+	AWSAccessValue []byte
+	AWSSecretValue []byte
+
+	// TODO: GCP and Azure
+
+	ResticPasswordValue []byte
+	ResticRepoValue     []byte
+)
+
+const (
+	resticSecretName = "restic-secret"
+)
+
+func (r *DataMoverBackupReconciler) CreateResticSecret(log logr.Logger) (bool, error) {
+
+	// get datamoverbackup from cluster
+	dmb := pvcv1alpha1.DataMoverBackup{}
+	if err := r.Get(r.Context, r.NamespacedName, &dmb); err != nil {
+		return false, err
+	}
+
+	// get pvc created by controller
+	pvcName := fmt.Sprintf("%s-pvc", dmb.Spec.VolumeSnapshotContent.Name)
+	pvc := corev1.PersistentVolumeClaim{}
+	if err := r.Get(r.Context, types.NamespacedName{Namespace: dmb.Namespace, Name: pvcName}, &pvc); err != nil {
+		return false, err
+	}
+
+	// define Restic secret to be created
+	newResticSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-secret", pvc.Name),
+			Namespace: dmb.Namespace,
+		},
+		Type: corev1.SecretTypeOpaque,
+	}
+
+	// Create Restic secret in OADP namespace
+	op, err := controllerutil.CreateOrUpdate(r.Context, r.Client, newResticSecret, func() error {
+		return r.buildResticSecret(newResticSecret, &dmb, &pvc)
```

Comment:
Suggestion: Add an owner reference to the secret. It will come in handy when we clean up
```
err := controllerutil.SetOwnerReference(&dmb, newResticSecret, r.Scheme)
			if err != nil {
				return err
			}
```